fc_util/src/lib.rs

```diff
@@ -3,57 +3,180 @@
 
 extern crate libc;
 
+use std::fmt;
+
 pub mod validators;
 
-pub fn timestamp_cycles() -> u64 {
-    #[cfg(target_arch = "x86_64")]
-    // Safe because there's nothing that can go wrong with this call.
-    unsafe {
-        std::arch::x86_64::_rdtsc() as u64
-    }
-    #[cfg(not(target_arch = "x86_64"))]
-    {
-        let mut ts = libc::timespec {
+const SECONDS_TO_NANOS: u64 = 1_000_000_000;
+
+/// Wrapper over libc::clockid_t to specify Linux Kernel clock source.
+pub enum ClockType {
+    /// Equivalent to libc::CLOCK_MONOTONIC.
+    Monotonic,
+    /// Equivalent to libc::CLOCK_REALTIME.
+    Real,
+    /// Equivalent to libc::CLOCK_PROCESS_CPUTIME_ID.
+    ProcessCpu,
+    /// Equivalent to libc::CLOCK_THREAD_CPUTIME_ID.
+    ThreadCpu,
+}
+
+/// Structure representing the date in local time with nanosecond precision.
+pub struct LocalTime {
+    /// Seconds in current minute.
+    sec: i32,
+    /// Minutes in current hour.
+    min: i32,
+    /// Hours in current day, 24H format.
+    hour: i32,
+    /// Days in current month.
+    mday: i32,
+    /// Months in current year.
+    mon: i32,
+    /// Years passed since 1900 BC.
+    year: i32,
+    /// Nanoseconds in current second.
+    nsec: i64,
+}
+
+impl LocalTime {
+    /// Returns the LocalTime structure for the calling moment.
```

Comment:
To make this show up nicely in the generated documentation:
```suggestion
    /// Returns the [`LocalTime`](struct.LocalTime.html) structure for the calling moment.
```